scripts/pylib/twister/twisterlib/hardwaremap.py

```diff
@@ -195,7 +195,7 @@ def discover(self):
                             self.options.platform.append(d.platform)
 
             elif self.options.device_serial:
-                self.add_device(self.options.device_serial,
+                self.add_device(self.options.device_serial[0],
```

Comment:
would this un-compatible with former settings? I mean we used to set the serial in hwmap as string?

Reply:
the first one:
```
- connected: true
  id: OLBZY05MJYIRV
  platform: mimxrt1060_evk@C/mimxrt1062/qspi
  product: linkserver
  runner: linkserver
  serial: /dev/ttyACM0
- connected: true
  id: "OLBZY05MJYIRV"
  platform: **unknown**
  product: Segger
  runner: jlink
  serial: /dev/ttyACM1
```
in second entry you should leave as was generated (`unknown`) or set the name of second core.
It is backward compatible.